benchmark/_driver.py

```diff
@@ -70,11 +70,11 @@ def benchmark(scenario, operation, metric, num_samples):
     :param IOperation operation: An operation to perform.
     :param IMetric metric: A quantity to measure.
     :param int num_samples: Number of samples to take.
-    :return: Deferred firing with a list of samples. Each sample is a
-        dictionary containing a ``success`` boolean. If ``success is True``,
-        the dictionary also contains a ``value`` for the sample measurement.
-        If ``success is False``, the dictionary also contains a ``reason`` for
-        failure.
+    :return: Deferred firing with a tuple containing one list of
+        benchmark samples and one scenario metrics result. See the
+        ``sample`` function for the structure of the samples.  The
+        scenario metrics are a dictionary containing information about
+        the scenario.
```

Comment:
:question: Alternatively, define a `PRecord` for samples and express this type using PEP-484 syntax. (I guess `Deferred[([[Sample], ScenarioMetricsResult)]`?)